router/batchrouter/asyncdestinationmanager/eloqua/bulk_uploader.go

```diff
@@ -0,0 +1,237 @@
+package eloqua
+
+import (
+	stdjson "encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+)
+
+func (b *EloquaBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+
+	file, err := os.Open(asyncDestStruct.FileName)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while opening the file. %v", err),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	defer file.Close()
+	eventType, customObjectId, err := checkEventType(file)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+			FailedReason:  fmt.Sprintf("got error while checking the event type. %v", err),
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+
+	fields := getFields(file)
+
+	var eloquaFields *Fields
+
+	if eventType == "track" {
+		customObjectData := Data{
+			BaseEndpoint:  b.baseEndpoint,
+			Authorization: b.authorization,
+			DynamicPart:   customObjectId,
+		}
+		eloquaFields, err = FetchFields(&customObjectData)
+		if err != nil {
+			return common.AsyncUploadOutput{
+				FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+				FailedReason:  fmt.Sprintf("got error while fetching fields. %v", err),
+				FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+				DestinationID: destination.ID,
+			}
+		}
+	} else {
+		customObjectData := Data{
+			BaseEndpoint:  b.baseEndpoint,
+			Authorization: b.authorization,
+		}
+		eloquaFields, err = FetchFields(&customObjectData)
+		if err != nil {
+			return common.AsyncUploadOutput{
+				FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
+				FailedReason:  fmt.Sprintf("got error while fetching fields. %v", err),
+				FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+				DestinationID: destination.ID,
+			}
+		}
+	}
+
+	filePAth, _ := createCSVFile(fields, file)
+
+	importDefinitionBody, err := createBodyForImportDefinition(eventType, fields, eloquaFields, file)
+	if err != nil {
+		return common.AsyncUploadOutput{
```

Comment:
[Minor] If below lines of code is common, we can write a common function which says `returnFailedOutput` with relevant arguments & call the function on every failure return
```golang
common.AsyncUploadOutput{
			FailedJobIDs:  append(asyncDestStruct.FailedJobIDs, asyncDestStruct.ImportingJobIDs...),
			FailedReason:  fmt.Sprintf("%v", err),
			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
			DestinationID: destination.ID,
		}
```